skip tests that are broken on NVIDIA Grace
- TestLinprogIPSparse tests are already skipped on some platforms, and IP is deprecated (https://github.com/scipy/scipy/pull/20522)
- sph_harm is deprecated (see https://github.com/scipy/scipy/pull/22167)
author: Kenneth Hoste (HPC-UGent)
--- scipy-1.16.0.orig/scipy/optimize/tests/test_linprog.py	2025-06-22 03:57:22.000000000 +0200
+++ scipy-1.16.0/scipy/optimize/tests/test_linprog.py	2025-06-26 08:22:33.743195292 +0200
@@ -2063,6 +2063,10 @@
         sys.platform == 'darwin',
         reason="Fails on macOS x86 Accelerate builds (gh-20510)"
     )
+    @pytest.mark.skipif(
+        platform.machine() == 'aarch64',
+        reason="Fails on NVIDIA Grace (Neoverse V2)"
+    )
     @pytest.mark.xfail_on_32bit("This test is sensitive to machine epsilon level "
                                 "perturbations in linear system solution in "
                                 "_linprog_ip._sym_solve.")
@@ -2117,6 +2121,10 @@
         sys.platform == 'darwin',
         reason="Fails on macOS x86 Accelerate builds (gh-20510)"
     )
+    @pytest.mark.skipif(
+        platform.machine() == 'aarch64',
+        reason="Fails on NVIDIA Grace (Neoverse V2)"
+    )
     @pytest.mark.xfail_on_32bit("This test is sensitive to machine epsilon level "
                                 "perturbations in linear system solution in "
                                 "_linprog_ip._sym_solve.")
--- scipy-1.16.0.orig/scipy/special/tests/test_sph_harm.py	2025-06-22 03:57:22.000000000 +0200
+++ scipy-1.16.0/scipy/special/tests/test_sph_harm.py	2025-06-26 09:14:10.278952180 +0200
@@ -1,3 +1,5 @@
+import platform
+
 import numpy as np
 import pytest
 
@@ -32,6 +34,10 @@
 
     @pytest.mark.parametrize("n_max", [7, 10, 50])
     @pytest.mark.parametrize("m_max", [1, 4, 5, 9, 14])
+    @pytest.mark.skipif(
+        platform.machine() == 'aarch64',
+        reason="Fails on NVIDIA Grace (Neoverse V2)"
+    )
     def test_all(self, n_max, m_max):
         theta = np.linspace(0, np.pi)
         phi = np.linspace(0, 2 * np.pi)
